File: ios/model/NewsMetadata.swift
import Foundation

struct NewsMetadata: Serializable {
    let uuid: String
    let title: String
    let timestamp: Date
    let image: String
    static func toObject(map: [String: Any]) -> NewsMetadata? {
        return map["uuid"] == nil || map["title"] == nil || map["timestamp"] == nil || map["image"] == nil ? nil : NewsMetadata(uuid: map["uuid"]! as! String, title: map["title"]! as! String, timestamp: DateService.toDate(from: map["timestamp"]! as! String), image: map["image"]! as! String)
    }
}

File: android/model/NewsMetadata.kt
package com.tobiasheymann.ep.model

import java.io.Serializable
import java.time.Instant

data class NewsMetadata(
    val uuid: String,
    val title: String,
    val timestamp: Instant,
    val image: String
) : Serializable {
    companion object {
        fun toObject(map: Map<String, Any>): NewsMetadata? {
            return if (
                map["uuid"] == null
                || map["title"] == null
                || map["timestamp"] == null
                || map["image"] == null
            )
                null
            else
                NewsMetadata(
                    map["uuid"]!! as String,
                    map["title"]!! as String,
                    Instant.parse(map["timestamp"]!! as String),
                    map["image"]!! as String
                )
        }
    }
}